feature/206

georgeenciu opened the pull request:
Resolves openscope/openscope#206 

from official charts found at www.aisro.ro 

[RNAV STAR CHART 08R/L](http://aisro.ro/aip/2017-02-02/DOCS/AIP/AD/AD2/AD_2_5_LROP/LR_AD_2_LROP_5-36_en.pdf)
[RNAV SID CHART 08R/L](http://aisro.ro/aip/2017-02-02/DOCS/AIP/AD/AD2/AD_2_5_LROP/LR_AD_2_LROP_5-34_en.pdf)
[RNAV STAR CHART 26R/L](http://aisro.ro/aip/2017-02-02/DOCS/AIP/AD/AD2/AD_2_5_LROP/LR_AD_2_LROP_5-37_en.pdf)
[RNAV SID CHART 26R/L](http://aisro.ro/aip/2017-02-02/DOCS/AIP/AD/AD2/AD_2_5_LROP/LR_AD_2_LROP_5-35_en.pdf)
[EN ROUTE CHARTS](http://aisro.ro/aip/2017-02-02/DOCS/AIP/ENR/ENR6/LR_ENR_6_2_en.pdf)

---

Depends on #33.

n8rzz commented:
@georgeenciu `v4.0.0` introduced `spawnPatterns` for airports.  please see the [spawnPatternReadme](https://github.com/openscope/openscope/blob/develop/src/assets/scripts/client/trafficGenerator/spawnPatternReadme.md) for more information on what, exactly, a `spawnPattern` is and how it should be structured.  Order of keys is important, is be sure to pay close attention to that as well.

Additionally, could you add a changelog entry as part of your PR?

Thanks!

georgeenciu commented:
@n8rzz Thanks for referencing the `spawnPattern`, will refactor the airport json accordingly
Will gladly create a changelog also, can you point to some instructions?

Thank you

n8rzz commented:
We already have a changelog, you'll just need to add an entry. This can be added to the `feature` section. https://github.com/openscope/openscope/blob/develop/CHANGELOG.md.  Just follow the same pattern of the previous entries.

indianbhaji commented:
@georgeenciu Hi, I'll take some time to review your pull request later. In the meantime, I've updated the title of the pull request and created #206. Title names are named after the issue they are based off. In this case, it is feature/206.

georgeenciu commented:
@indianbhaji actually this is `LROP Bucharest Henri Coanda International Airport` in Romania and not `LHBP Budapest Ferenc Liszt International Airport` in Hungary :)

do you need me to change the arrivals and departures to spawnPattern before the review?

n8rzz commented:
@georgeenciu you will need `spawnPatterns` in order to add aircraft into the sim.  I've spun up a review app so you can view your work.  Anytime you push new code to your branch, the review app should update (but it takes a few minutes to rebuild).

You can always run locally, which will speed things up significantly, but also assumes basic knowledge of web systems and front end technologies.  (see: https://github.com/openscope/openscope/tree/develop/tools for setup info).

indianbhaji commented:
@georgeenciu oops, you didn't see that

georgeenciu commented:
@indianbhaji don't beat yourself up, Metallica did the same mistake in a live concert in front of thousands

indianbhaji commented on assets/airports/airportLoadList.js:
```diff
@@ -257,6 +257,12 @@ window.AIRPORT_LOAD_LIST = (function() { // eslint-disable-line wrap-iife
+            icao: 'lrop',
+            level: 'medium',
+            name: 'Bucharest Henri Coanda International Airport',
+            wip: false
```
I believe that `wip` should be set to `true` whilst it's being reviewed, or something like that @erikquinn can correct me if I'm wrong!

indianbhaji replied:
I believe that `wip` should be set to `true` whilst it's being reviewed, or something like that @erikquinn can correct me if I'm wrong!

erikquinn replied:
Yessir! Only the most fully refined airports will have a `wip: false`. See #179 for more on this.

indianbhaji commented on assets/airports/lrop.json:
```diff
@@ -0,0 +1,674 @@
+        {
+            "name":        ["08R", "26L"],
+            "name_offset": [[0, 0], [0, 0]],
+            "length": 3.5,
```
The `length` attribute is deprecated as it is now calculated from the start and end coordinates of the runway

indianbhaji commented on assets/airports/lrop.json:
```diff
@@ -0,0 +1,674 @@
+    "runways": [
+        {
+            "name":        ["08R", "26L"],
+            "name_offset": [[0, 0], [0, 0]],
```
The `name_offset` is unnecessary here as there is no offset.

indianbhaji commented on assets/airports/lrop.json:
```diff
@@ -0,0 +1,674 @@
+        },
+        {
+            "name":        ["08L", "26R"],
+            "name_offset": [[0, 0], [0, 0]],
```
This `name_offset` is also unnecessary. The only attributes that are required are `name`, `end` and `ils`

indianbhaji commented on assets/airports/lrop.json:
```diff
@@ -0,0 +1,674 @@
+            "suffix": {"08R":"", "08L":""},
+            "rwy": {  
+                "08R" : ["OP888"],
+                "08L" : [ "OP887"]
```
Unnecessary space (it's bugging me, don't ask!)

indianbhaji commented on assets/airports/lrop.json:
```diff
@@ -0,0 +1,674 @@
+    		"origin": "LROP",
+            "destination": "",
+            "category": "departure",
+            "route": "LROP.DENAK1K.KODRU",
```
Should be `LROP.DENAK1K.DENAK` as detailed in your SIDs section, as your exitPoint is called `DENAK`

georgeenciu replied:
correct! i was trying to see if that works and make the planes go to an exit point of the country rather than an exit point to the TMA

erikquinn replied:
@georgeenciu The only restriction, as @indianbhaji pointed out, is that in a route of `ENTRY.PROCEDURE.EXIT`, the value of `ENTRY` and `EXIT` must be defined in the appropriate `entryPoints` or `exitPoints` section in that SID or STAR.

If you want the aircraft spawning on this pattern to "go to an exit point of the country", you can also do that by continuing the route, for example: `ENTRY.PROCEDURE.EXIT..FIXXA..FIXXB..FIXXC`.

indianbhaji commented on assets/airports/lrop.json:
```diff
@@ -0,0 +1,674 @@
+            "category": "arrival",
+            "route": "ARGES.OSTAL1U.LROP",
+            "altitude":  [15000, 17000],
+            "speed": 350,
```
This `speed` may be too fast for the aircraft in the `ROT` fleet, such as the AT45. 320 should do fine!

erikquinn replied:
My rules of thumb with speeds:
 - if it doesn't matter, set it to 280 (very unlikely you'll run into problems except with light GA)
 - if turboprops are involved, 250 may be necessary
 - if it's all turbojets, you can go as high as 320

Do with that what you will. You can also feel free to look up the list of what aircraft types will spawn, and check their max speeds, etc, in order to find out how fast you can ask them to spawn before some aircraft will be unable to comply.

indianbhaji commented on assets/airports/lrop.json:
```diff
@@ -0,0 +1,674 @@
+            "category": "arrival",
+            "route": "DENAK.DENAK1U.LROP",
+            "altitude":  [15000, 17000],
+            "speed": 350,
```
This `speed` may be too fast for the aircraft in the `ROT` fleet, such as the AT45. 320 should do fine!

indianbhaji commented on assets/airports/lrop.json:
```diff
@@ -0,0 +1,674 @@
+            "category": "arrival",
+            "route": "NETUL.NETUL2U.LROP",
+            "altitude":  [17000, 20000],
+            "speed": 350,
```
This `speed` may be too fast for the aircraft in the `ROT` fleet, such as the AT45. 320 should do fine!

indianbhaji commented on assets/airports/lrop.json:
```diff
@@ -0,0 +1,674 @@
+            "category": "arrival",
+            "route": "IDARU.IDARU2U.LROP",
+            "altitude":  [17000, 20000],
+            "speed": 350,
```
This `speed` may be too fast for the aircraft in the `ROT` fleet, such as the AT45. 320 should do fine!

indianbhaji commented on assets/airports/lrop.json:
```diff
@@ -0,0 +1,674 @@
+            "category": "arrival",
+            "route": "DENAK.DENAK1U.LROP",
+            "altitude":  [15000, 17000],
+            "speed": 350,
```
This `speed` may be too fast for the aircraft in the `ROT` fleet, such as the AT45. 320 should do fine!

indianbhaji commented on assets/airports/lrop.json:
```diff
@@ -0,0 +1,674 @@
+            "category": "arrival",
+            "route": "SOMOV.SORDU1U.LROP",
+            "altitude": [14000, 17000],
+            "speed": 350,
```
This `speed` may be too fast for the aircraft in the `ROT` fleet, such as the AT45. 320 should do fine!

indianbhaji commented on assets/airports/lrop.json:
```diff
@@ -0,0 +1,674 @@
+            "name": "Sordu One Uniform",
+            "suffix": {"08R":"", "08L":""},
+            "entryPoints": {
+                "SOMOV": ["SOMOV" ],
```
Extra space

n8rzz commented:
@georgeenciu also be sure to pull in develop because that has been updated since you last pushed.

erikquinn commented on assets/airports/lrop.json:
```diff
@@ -0,0 +1,670 @@
+        "SOKRU1K": { 
+	        "icao": "SOKRU1K",       
+	        "name": "Sokru One Kilo",
+	        "suffix": {"08R":"", "08L":""},
```
Work on #33 _will_ be happening soon enough, so the suffix key will actually be used in time, so I'd like to start building new airports in accordance with that design. So the way that will work is the procedure will be called `SOKRU`, and the `suffix` for each runway will be listed here.

georgeenciu replied:
@erikquinn `LROP` is a bit special from this point of view, as it has the same procedure for both 08L and 08R, meaning `SOKRU` departure  will look like this: `"suffix": {"08R":"1K", "08L":"1K", "26L": "1M", "26R": "1M"}` and same goes for the arrivals 
will this be a problem?

indianbhaji replied:
@georgeenciu That's fine

georgeenciu replied:
`TOSVI` arrival for 26 L/R has 2 separate procedures in case of high traffic `TOSVI3X` and `TOSVI1R`, how will I specify 2 procedures for same runway coming from the same entry point?

erikquinn commented on assets/airports/lrop.json:
```diff
@@ -0,0 +1,670 @@
+	        "rwy": {
+	            "08R" : [["OP882", "S220-"], "OP890"],
+	            "08L" : [["OP881", "A09+|S220-"], "OP890"] 
+	        },
```
Verify the `SOKRU` departure is not valid from runways 26L/26R? If not listed here, the procedure cannot be used by an aircraft who departs/departed a runway not specified here.

georgeenciu replied:
@erikquinn airport file has only the sids and stars for 08L/R, will add soon the procedures for 26L/R

erikquinn commented on assets/airports/lrop.json:
```diff
@@ -0,0 +1,670 @@
+            "rate": 1,
+            "airlines": [
+                ["ROT", 1]
+				
```
remove empty line

erikquinn commented on assets/airports/lrop.json:
```diff
@@ -0,0 +1,670 @@
+                           ["N44.564661", "E26.076747", "314ft"],
+                           ["N44.56797", "E26.120428", "314ft"]
+                         ],
+            "delay":       [2, 2],
```
this `delay` property is not in use; please remove all use of this parameter.

georgeenciu commented on assets/airports/lrop.json:
```diff
@@ -0,0 +1,670 @@
+                           ["N44.56797", "E26.120428", "314ft"]
+                         ],
+            "delay":       [2, 2],
+            "sepFromAdjacent": [2.5, 2.5],
```
@erikquinn "Distance in nautical miles that another aircraft can come while on parallel approach paths, a violation will occur at 85% of this value" 
is this not in use anymore?

georgeenciu commented:
- added all 26R/L procedures and changed the `sids` and `stars` to comply with the future changes in #33 
- added a separate TOSVI1R arrival with no suffixes 
`"TOSVI": {"suffix": {"08R": "2U", "08L": "2U", "26R": "3X", "26L": "3X"},...}`
`"TOSVI1R": {"suffix": {"26R": "", "26L": ""},...}`
- waiting for work to finish on #33 so this actually works :)

erikquinn commented:
@georgeenciu We always want to see our tests run on any code that is to be merged into the sim, and we currently aren't able to run these tests on external forks.

I've added you to the [openscope team](https://github.com/orgs/openscope/teams/simulator-collaborators), just check your email to accept the invitation. Then, you will have access to push directly to the branch named `feature/206` on the `openscope` repository, where I have moved all your changes over to. The month of May will be all about airport changes, so that's when you can expect to see it further critiqued, amended, and merged. 😄

Feel free to join the [openscope slack team](http://slack.openscope.co) as well! That's where we all coordinate what we're doing, and there's always someone around to provide help with any issues you may encounter.

erikquinn commented:
Closed in favor of local branch's PR, #336.
